### PR TITLE
Update tutorials

### PR DESCRIPTION
**Pull request by pyrocto**
Fixed some typos in the mercury tutorial and rewrote a copy for the 0.1 compiler.  Added files to the examples directory for each of the samples in the tutoral.

### CODE REVIEW

File: rholang/examples/tut-iterate.rho [hunk 1]
```diff
@@ -0,0 +1,39 @@
+new channel, iterate in {
```
Comment (dckc) [inline rholang/examples/tut-iterate.rho]: When I run the result of compiling this one, I get:

```
rosette> *** formals mismatch:
 formals = [[channel unification_ptrn product]]
 actuals = [['Rholang-iterate-165203013 'wildcard [4 5 6] 'Rholang-channel--307376460]]
 in (FOp  produce), on (FO NameSpaceSBO 151056708)
```

File: rholang/examples/tut-philosophers.rho [hunk 2]
```diff
@@ -0,0 +1,14 @@
+new north, south, knife, spoon in {
+    north!(knife) |
+    south!(spoon) |
+    for (knf <- north; spn <- south) {
+        "Philosopher 1 Utensils: ".display(knf, ", ", spn, "\n") |
```
Comment (dckc) [inline rholang/examples/tut-philosophers.rho]: this looks like strings have a display method; i.e. that every string carries the capability to print itself. Am I reading that correctly?

File: rholang/examples/tut-philosophers.rho [hunk 3]
```diff
@@ -0,0 +1,14 @@
+new north, south, knife, spoon in {
```
Comment (dckc) [inline rholang/examples/tut-philosophers.rho]: This example seems to work:

```
rosette> Philosopher 1 Utensils: Rholang-knife--157425684, Rholang-spoon--376710479
Philosopher 2 Utensils: Rholang-knife--157425684, Rholang-spoon--376710479
```

File: rholang/examples/tut-hello-again.rho [hunk 4]
```diff
@@ -0,0 +1,12 @@
+new helloAgain in {
+  contract helloAgain(_) = {
+    new chan in {
+      chan!("Hello again, world!") |
```
Comment (dckc) [inline rholang/examples/tut-hello-again.rho]: missing a newline here?

The REPL output I get via the web compiler is:

```
rosette> Hello again, world!Hello again, world![]
rosette> 
```

File: rholang/examples/tut-iterate.rho (hunk 1 above)
Comment (dckc) [reply rholang/examples/tut-iterate.rho]: the rbl code I see is:

```
(let [[channel (generateFresh "channel")] [iterate (generateFresh "iterate")]] (block (block ( (proc [] (run (compile (let [[[[Rholangf908bcef0c1a] [[Rholang1e215af8ff24 Rholang7276f059138a]]] (consume t [iterate] [**wildcard**] [['list 'channel]] #t)]] ( (proc [[list channel]] (let [[ack (generateFresh "ack")] [right (generateFresh "right")]] (block (produce t channel **wildcard** ack) (let [[[[Rholangcaef8a297743] [Rholangd33771e6ce9d]] (consume t [ack] [**wildcard**] ['**wildcard**] #f)]] ( (proc [[**wildcard**]] (block ( (proc [] (run (compile (let [[[[Rholang116f1fc557a8] [Rholangf58992db16a4]] (consume t [right] [**wildcard**] ['pair] #t)]] ( (proc [pair] (if (match? t pair [i limit]) ( (proc [[i limit]] (block (produce t channel **wildcard** [( nth list i) (< i limit)]) (let [[[[Rholang0d4fa50d1260] [Rholang8a8e4315b8f6]] (consume t [ack] [**wildcard**] ['**wildcard**] #f)]] ( (proc [[**wildcard**]] (if (match? t (< (+ i 1) limit) #t) (produce t right **wildcard** [(+ i 1) limit]) #niv)) [Rholang8a8e4315b8f6])))) pair) #niv)) Rholangf58992db16a4)))))) (produce t right **wildcard** [0 ( size list)]))) [Rholangd33771e6ce9d]))))) [Rholang1e215af8ff24 Rholang7276f059138a])))))) (produce t iterate **wildcard** [4 5 6] channel)) (let [[[[Rholang281e196673d8] [Rholangf7afee988bae]] (consume t [channel] [**wildcard**] ['ack] #f)]] ( (proc [[ack]] (block (produce t ack **wildcard** #niv) (let [[left (generateFresh "left")]] (block ( (proc [] (run (compile (let [[[[Rholang2bf8ab3e7e4d] [Rholangc579826adaca]] (consume t [left] [**wildcard**] ['**wildcard**] #t)]] ( (proc [**wildcard**] (let [[[[Rholang37e61825bd3b] [Rholangab2d333f0c1e]] (consume t [channel] [**wildcard**] ['pair] #f)]] ( (proc [[pair]] (if (match? t pair [v keepGoing]) ( (proc [[v keepGoing]] (block ( display v " ") (if (match? t keepGoing #t) (block (produce t ack **wildcard** #niv) (produce t left **wildcard** #niv)) #niv))) pair) #niv)) [Rholangab2d333f0c1e]))) Rholangc579826adaca)))))) (produce t left **wildcard** #niv))))) [Rholangf7afee988bae]))))
```

File: rholang/examples/tut-make-cell.rho [hunk 5]
```diff
@@ -0,0 +1,37 @@
+new MakeCell in {
```
Comment (dckc) [inline rholang/examples/tut-make-cell.rho]: losing here too:

```
*** formals mismatch:
 formals = [[channel unification_ptrn product]]
 actuals = [['Rholang-MakeCell--261298636 'wildcard 123 'Rholang-myGet-47677321 'Rholang-mySet--32894946]]
 in (FOp  produce), on (FO NameSpaceSBO 144453956)
```

rbl code:
```
(let [[MakeCell (generateFresh "MakeCell")]] (block ( (proc [] (run (compile (let [[[[Rholang819262138c50] [[Rholange097047ec0d8 Rholang267b530df618 Rholang974a4d24260e]]] (consume t [MakeCell] [**wildcard**] [['init 'get 'set]] #t)]] ( (proc [[init get set]] (let [[valueStore (generateFresh "valueStore")]] (block (block (produce t valueStore **wildcard** init) ( (proc [] (run (compile (let [[[[Rholangf433b954415a] [Rholang7b4a6cd05fdf]] (consume t [get] [**wildcard**] ['ack] #t)]] ( (proc [ack] (let [[[[Rholangb042e3c54d23] [Rholang2e4c67045015]] (consume t [valueStore] [**wildcard**] ['value] #f)]] ( (proc [[value]] (block (produce t valueStore **wildcard** value) (produce t ack **wildcard** value))) [Rholang2e4c67045015]))) Rholang7b4a6cd05fdf))))))) ( (proc [] (run (compile (let [[[[Rholang0e9a428569a8] [Rholangd905ffa9c4e8]] (consume t [set] [**wildcard**] ['pair] #t)]] ( (proc [pair] (let [[[[Rholang8f89b812e5d4] [Rholang8a20ce843cd0]] (consume t [valueStore] [**wildcard**] ['**wildcard**] #f)]] ( (proc [[**wildcard**]] (if (match? t pair [newValue ack]) ( (proc [[newValue ack]] (block (produce t valueStore **wildcard** newValue) (produce t ack **wildcard** #niv))) pair) #niv)) [Rholang8a20ce843cd0]))) Rholangd905ffa9c4e8))))))))) [Rholange097047ec0d8 Rholang267b530df618 Rholang974a4d24260e])))))) (let [[myGet (generateFresh "myGet")] [mySet (generateFresh "mySet")]] (block (produce t MakeCell **wildcard** 123 myGet mySet) (let [[ack (generateFresh "ack")]] (block (produce t myGet **wildcard** ack) (let [[[[Rholangd6a9594635c9] [Rholangb8367d3166be]] (consume t [ack] [**wildcard**] ['result] #f)]] ( (proc [[result]] (block (block ( display result) (produce t mySet **wildcard** [456 ack])) (let [[[[Rholang1a1a85a09229] [Rholang64b6cd529539]] (consume t [ack] [**wildcard**] ['**wildcard**] #f)]] ( (proc [[**wildcard**]] (block (produce t myGet **wildcard** ack) (let [[[[Rholang8a6d49cc356b] [Rholanga5ea389d9f46]] (consume t [ack] [**wildcard**] ['result] #f)]] ( (proc [[result]] ( display result)) [Rholanga5ea389d9f46])))) [Rholang64b6cd529539])))) [Rholangb8367d3166be]))))))))
```

**Comment by guardbotmk3** [comment]
Please hold off. Mike is writing these relative to my PR fixing the release
blockers. If you're really eager, try pulling that branch.

File: rholang/examples/tut-hello-again.rho (hunk 4 above)
Comment (dckc) [reply rholang/examples/tut-hello-again.rho]: never mind; that's addressed by guardbotmk3:dev-fix-release-blockers

File: rholang/examples/tut-iterate.rho (hunk 1 above)
Comment (dckc) [reply rholang/examples/tut-iterate.rho]: also addressed by guardbotmk3:dev-fix-release-blockers

File: rholang/examples/tut-make-cell.rho (hunk 5 above)
Comment (dckc) [reply rholang/examples/tut-make-cell.rho]: Now that I merged guardbotmk3:dev-fix-release-blockers, it runs, but a newline might be nice:

```
rosette> 123456[]
rosette> 
```

**Comment by pyrocto** [comment]
Dan, I'm so sorry.  I should have made it clear that this change depends on #188.

**Comment by pyrocto** [comment]
Added "\n" in a few places.  Please reapprove.  Thanks!

**Pull request merged**